### PR TITLE
Updating README & removing previous CLA

### PR DESCRIPTION
**Pull request by callum-mcdata**
## PR Notes
- Removed the old Transform CLA as we've installed `cla-bot` that links out to the dbt labs CLA.
- Updated links in the README to point towards the new repo location & dbt labs (where it made sense)

### What the PR doesn't do
I didn't update the links to the documentation or the package installation sections, as those are still things that we will address in the future.

### CODE REVIEW

File: README.md
Comment (tlento) [inline README.md:2]: Do we want to change this yet? I think we'd want this to continue linking to wherever the MetricFlow docs are housed.

Ideally we'd get things merged in under https://www.getdbt.com/product/semantic-layer/ or something, but that doesn't seem to have happened yet.

File: README.md
Comment (tlento) [inline README.md:13]: Hm, ideally we'd be able to link to the metricflow-under-dbt community Slack whenever that materializes.

File: README.md
Comment (tlento) [inline README.md:85]: same question

**Review by tlento — APPROVED** [review]
Thanks for doing this! I'm not too fussed about it, but we might want to hold off on linking to the dbt pages for things that don't incorporate metricflow yet. 

It's your call - if you want to revert those few lines feel free to do it and merge right after.

File: README.md
Comment (callum-mcdata) [reply README.md:13]: We have channels within the dbt slack community that will be dedicated to MetricFlow but that doesn't exist quite yet. I will make sure to update this link in the future when it does

File: README.md
Comment (callum-mcdata) [reply README.md:2]: Fair callout - I'll remove this and we can update it once we've updated the top level org links.

**Pull request merged**
